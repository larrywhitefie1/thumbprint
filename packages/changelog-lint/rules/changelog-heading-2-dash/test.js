const test = require('../../test-utils');
const rule = require('./');

const valid = [
    '## Unreleased',
    '## 1.3.2 - 2010-02-02',
    `## Unreleased

## 1.3.2 - 2010-02-02`,
];

const invalid = [
    '## 1.3.2 -2010-02-02',
    '## 1.3.2-2010-02-02',
    '## 1.3.2 -  2010-02-02',
    '## 1.3.2  -  2010-02-02',
    '## 1.3.2 â€” 2010-02-02',
    '## 1.3.2 -- 2010-02-02',
    '## 1.3.2 2010-02-02',
];

test({
    rule,
    valid,
    invalid,
});
