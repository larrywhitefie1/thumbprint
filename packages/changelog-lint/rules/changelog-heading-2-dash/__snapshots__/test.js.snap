// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid case #0 fails 1`] = `
Object {
  "input": "## 1.3.2 -2010-02-02",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 21,
          "line": 1,
          "offset": 20,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Second level heading contains a dash “-“ separated by one space on each side.",
      "ruleId": "changelog-heading-2-dash",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #1 fails 1`] = `
Object {
  "input": "## 1.3.2-2010-02-02",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Second level heading contains a dash “-“ separated by one space on each side.",
      "ruleId": "changelog-heading-2-dash",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #2 fails 1`] = `
Object {
  "input": "## 1.3.2 -  2010-02-02",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 23,
          "line": 1,
          "offset": 22,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Second level heading contains a dash “-“ separated by one space on each side.",
      "ruleId": "changelog-heading-2-dash",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #3 fails 1`] = `
Object {
  "input": "## 1.3.2  -  2010-02-02",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 24,
          "line": 1,
          "offset": 23,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Second level heading contains a dash “-“ separated by one space on each side.",
      "ruleId": "changelog-heading-2-dash",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #4 fails 1`] = `
Object {
  "input": "## 1.3.2 — 2010-02-02",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 22,
          "line": 1,
          "offset": 21,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Second level heading contains a dash “-“ separated by one space on each side.",
      "ruleId": "changelog-heading-2-dash",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #5 fails 1`] = `
Object {
  "input": "## 1.3.2 -- 2010-02-02",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 23,
          "line": 1,
          "offset": 22,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Second level heading contains a dash “-“ separated by one space on each side.",
      "ruleId": "changelog-heading-2-dash",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #6 fails 1`] = `
Object {
  "input": "## 1.3.2 2010-02-02",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Second level heading contains a dash “-“ separated by one space on each side.",
      "ruleId": "changelog-heading-2-dash",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;
